rust/oak/src/grpc/mod.rs

```diff
@@ -176,3 +176,53 @@ pub fn event_loop<T: OakNode>(mut node: T, grpc_in_handle: ReadHandle) -> i32 {
         );
     }
 }
+
+pub fn handle_req_rsp<C, R: protobuf::Message, Q: protobuf::Message>(
+    mut node_fn: C,
+    req: &[u8],
+    mut writer: ChannelResponseWriter,
+) where
+    C: FnMut(R) -> Result<Q>,
+{
+    let r: R = protobuf::parse_from_bytes(&req).unwrap();
+    match node_fn(r) {
+        Ok(rsp) => writer.write(rsp, WriteMode::Close),
+        Err(status) => writer.close(Err(status)),
+    }
+}
+
+pub fn handle_req_stream<C, R: protobuf::Message>(
```

Comment:
That's probably helpful, but I don't think it will belong at this level &ndash; I'm expecting there to be a higher level framework built on this that will encapsulate all sorts of handy async things (and that's covered by the planning docs elsewhere).